File: 800/AmusingJoke.py
s1=input()
s2=input()
s3=input()
l1=[x for x in s1]
l2=[x for x in s2]
l3=[x for x in s3]
l4=l1+l2
y=1
for i in l3:
    if l3.count(i)==l4.count(i):
        continue
    else:
        print("NO")
        y=2
        break
if y!=2:
    print("YES")

File: 800/AmusingJoke.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    string s1,s2,s3;
    cin>>s1>>s2>>s3;
    string combined=s1+s2;

     // Count characters
     // map stores key value pair. each key is a character and its value is the 
     // count of that character in the combined string
    map<char,int> count;
    for (char ch:combined)
        count[ch]++;

    // for each appearance in third string we subtract one from char count in map
    // if at the end all are 0 then their counts match in combined and s3 strings
    for (char ch:s3)
        count[ch]--;

     // Check if all counts are zero
    for (auto [ch, val] : count) {
        if (val != 0) {
            cout << "NO\n";
            return 0;
        }
    }
    cout << "YES\n";
    
    return 0;
}